### DISCUSSION TITLE
Supporting Gossamer for the spawn phase

### DISCUSSION THREAD

**Discussion by qdm12** [paritytech/zombienet#623] [General]
Hi there 👋

I'm Quentin, part of the team working on Gossamer, the Go Polkadot host.

Our team has been eyeing Zombienet for a few months now, and me and Ed from our team attended recently the Zombienet talk by Javier at Sub0 Lisbon.

This made us realise we had a bunch of Go code and docker-compose configurations aiming to do what Zombienet does (but so far behind, obviously!).
The solution seems obvious: we need Zombienet to be able to spawn Gossamer nodes ⭐ That would help us tremendously improve our CI and testing strategy.

There are a few questions on that:

1. How does Zombinet configure Substrate nodes? We probably need to harmonize configuration flags/files between the two, we can do modifications to Gossamer or we can have translations in Zombienet 🤔
2. Gossamer doesn't handle Parachains for now, so we would have to disable this in Zombienet for Gossamer nodes
3. We believe the testing phase of Zombienet should work as-is with Gossamer (it has logs, an rpc server matching the spec), do you also think this is the case?
4. Side question: do you support other hosts such as Smoldot already?

Thank you!

**Comment by pepoviola** [comment]
Hi @qdm12,
                     Thanks for your feedback and open this thread, related to your questions.
                     
> 1. How does Zombinet configure Substrate nodes? We probably need to harmonize configuration flags/files between the two, we can do modifications to Gossamer or we can have translations in Zombienet 🤔

Zombienet at the moment allow you to configure how to run your node in a couple of ways:
  - You can define the `command` to run the node and the `arguments` to use, but in this case Zombienet will add this args by default
  
  ```
      --rpc-cors all --unsafe-rpc-external --rpc-methods unsafe --unsafe-ws-external
  ```   
  And the reset of the args can be managed by config keys in your network topology file (toml/json)
  
  - You can also define the command __and__ the arguments to run with `command_with_args`, in this case Zombienet will use the supplied configuration to run the node.
  

> 2. Gossamer doesn't handle Parachains for now, so we would have to disable this in Zombienet for Gossamer nodes

Yes, you can define only the relaychain and works as expected.

> We believe the testing phase of Zombienet should work as-is with Gossamer (it has logs, an rpc server matching the spec), do you also think this is the case?

Yes, I have to make a PoC running `Gossamer` nodes but if the node expose an rpc server and metrics through prometheus should works as expected.

> Side question: do you support other hosts such as Smoldot already?                     

At the moment we didn't test with Smoldot as host, but I think should works (ping @wirednkod). As side node, substrate-connect use Zombienet in their CI.

Again, thanks for your feedback and we can help you to setup a PoC using Zombienet and Gossamer. Please ping us if you have questions.

Thanks!
